h2o-core/src/main/java/water/JettyHTTPD.java

```diff
@@ -117,13 +118,25 @@ protected void registerHandlers(HandlerWrapper handlerWrapper, ServletContextHan
     context.addServlet(PutKeyServlet.class,   "/3/PutKey");
     context.addServlet(RequestServer.class,   "/");
 
+
+    final List<Handler> extHandlers = new ArrayList<Handler>();
+    extHandlers.add(new AuthenticationHandler());
+    // here we wrap generic handlers into jetty-aware wrappers
+    for (final ExtensionHandler extensionHandler : ExtensionHandler.REGISTRY.getAll()) {
```

Comment:
is there a reason why ExtensionHandler is not a Handler directly? Why do we need to wrap it like this into AbstractHandler?

Reply:
because `Handler` is a Jetty class and we need to avoid direct dependency on Jetty

Reply:
Okay, makes sense!

Maybe the handle() method can return `int` (response status) - we will also need to replace the call to `setResponseStatus` in the proprietary extension.

Reply:
not really - `setResponseStatus` is a static helper, independent of Jetty, and as such it was (will have been) extracted from the Jetty-dependent code. So, if isolation from Jetty is the only reason why you suggest it, it is not necessary and I would rather leave it this way, i.e. using the return value to isolate the call to jetty-specific `baseRequest.setHandled(true)` method.